From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tariq Bashir <120014322+t-bashir-bs@users.noreply.github.com>
Date: Tue, 28 May 2024 15:34:10 +0100
Subject: OS-17053: Use same nssdb path for roKeyStore and Chromium

This change has been cherry picked from the QtWebEngine patch
I94eb625e805f9b2fe19c85f8f574f5725f0dbd46.

roKeyStore creates/uses a nssdb store in /var/run/.pki/nssdb, whereas
Chromium expects it to be in user home directory, which is not writable
for us.

Modified Chromium source code to use the same nssdb as roKeyStore.

diff --git a/crypto/nss_util.cc b/crypto/nss_util.cc
index bfd752890aa73d43343590c85cdc14abcbbd6c3e..8ba42723a657f836422d484c887cb78fd628211d 100644
--- a/crypto/nss_util.cc
+++ b/crypto/nss_util.cc
@@ -44,9 +44,9 @@ base::FilePath GetXdgDataNssdbDirectory() {
 }
 
 base::FilePath GetHomeNssdbDirectory() {
-  base::FilePath home_dir;
-  base::PathService::Get(base::DIR_HOME, &home_dir);
+  base::FilePath home_dir("/var/run");
   if (home_dir.empty()) {
+    LOG(ERROR) << "Failed to get home directory /var/run.";
     return {};
   }
 
